setup.py

```diff
@@ -749,14 +750,18 @@ def is_package_excluded(package: str, exclusion_list: List[str]):
 def do_setup():
     """Perform the Airflow package setup."""
     write_version()
+    # Note! Exclude does not work when combined with 'airflow' include so we need to remove
+    # providers manually
+    all_packages = [package for package in find_packages(include=['airflow', 'airflow.*'])
+                    if not package.startswith('airflow.providers')]
     setup(
         name='apache-airflow',
         description='Programmatically author, schedule and monitor data pipelines',
         long_description=long_description,
         long_description_content_type='text/markdown',
         license='Apache License 2.0',
         version=version,
-        packages=find_packages(include=['airflow', 'airflow.*']),
+        packages=all_packages,
```

Comment:
```suggestion
        packages=find_packages(include=['airflow*'], exclude=['airflow.providers', 'airflow.providers.*']),
```

Tested with:

```
[package for package in find_packages(include=['airflow', 'airflow.*']) if not package.startswith('airflow.providers')] == find_packages(include=['airflow*'], exclude=['airflow.providers', 'airflow.providers.*'])
```

Reply:
Testing. The setup tools behaviour is often tricky :)

Reply:
One thing (but it is already covered in our release docs) is to clean up the egginfo :(

Reply:
BTW. I stil have to test all the behaviour in Dockerfile/local installation with this setup.py - i.e. how people will be testing it locally. I believe all works fine for pip install -e . and when I add extras -> providers links it will work as well for the Dockerfile But this can be done later.

Reply:
Confrmed. Works fine with this exclude :)